packages/apollo/src/commands/service/check.ts

```diff
@@ -271,6 +273,7 @@ export default class ServiceCheck extends ProjectCommand {
 
           const configName = config.name;
           const tag = flags.tag || config.tag || "current";
+          const serviceName = flags.serviceName;
```

Comment:
I think `config.name` (above) and `flags.serviceName` are the same, but from different locations. I think we should resolve similarly to how we resolve tag, with the flag overriding the value from config if it's present.

Reply:
I may be mistaken here, as there might be some implicit federation logic that differentiates this. If that's the case, could you please add a comment explaining what's going on here?

Reply:
`config.name` is the graph name (`engine`) and `serviceName` is the implementing service name (`accounts`). Great place to use some comments and more descriptive naming.